src/scenes/AcceleratorRouter/Documents/DocumentView/DocumentVariablesTab.tsx

```diff
@@ -159,22 +158,21 @@ const DocumentVariablesTab = ({ projectId: projectIdProp, setSelectedTab }: Docu
     );
   }, [documentSectionVariables, getUsedSections, variables]);
 
-  const onSectionClicked = useCallback(
-    (sectionNumber: string) => {
-      setOpenEditVariableModal(false);
-      setSelectedVariable(undefined);
-
-      if (setSelectedTab) {
-        setSelectedTab(strings.DOCUMENT);
-        setTimeout(() => {
-          // do find and scroll async to allow tab-switching to occur first
-          const relevantSection = document.getElementById(sectionNumber);
-          relevantSection?.scrollIntoView({ behavior: 'smooth' });
-        }, 0);
-      }
-    },
-    [setSelectedTab]
-  );
+  const onSectionClicked = useCallback((sectionNumber: string) => {
+    setOpenEditVariableModal(false);
+    setSelectedVariable(undefined);
+
+    const documentTabElement: HTMLButtonElement | null = document.querySelector('button[role="tab"]:first-child');
+    if (documentTabElement) {
+      documentTabElement.click();
```

Comment:
I think this is my only qualm with this PR - the `querySelector` and `first-child`. And I don't necessarily know of a better way to do it. 
The only other thing I can think of is doing a `navigate` with `replace: true` and just modifying the query params. Not sure if that would be considered better or worse though

Reply:
I'm not a fan of this either, but it eliminates the circular dependency and we're already accessing the DOM within this function. The selector still works without `:first-child` in the selector, I added it to make it more specific. But perhaps it's too specific, will remove it for now.

Reply:
I agree this can be cleaned up further. I'll go ahead and merge this to prevent merge conflicts in the PR and to fix the back button behavior across Terraware. And I'll try to circle back today to improve this code, but I also wouldn't be opposed to you @ice1080 or @constanzauanini making the improvements if you get the itch to.

Thanks for not holding up this PR over this change 🙌

Reply:
Oh definitely, and this isn't the only usage of `document.querySelector` anyway, and also I feel it should generally be treated as a guideline and not a hard rule. I would say don't worry about it, I just wanted to mention it.